provider_client.go

```diff
@@ -226,3 +229,72 @@ func (client *ProviderClient) Request(method, url string, options RequestOpts) (
 
 	return resp, nil
 }
+
+func defaultOkCodes(method string) []int {
+	switch {
+	case method == "GET":
+		return []int{200}
+	case method == "POST":
+		return []int{201, 202}
+	case method == "PUT":
+		return []int{201, 202}
+	case method == "DELETE":
+		return []int{202, 204}
+	}
+
+	return []int{}
+}
+
+func (client *ProviderClient) Get(url string, JSONResponse *interface{}, opts *RequestOpts) (*http.Response, error) {
+	if opts == nil {
+		opts = &RequestOpts{}
+	}
```

Comment:
Any reason to handle the `opts == nil` case in these utility methods instead of in `Request` itself? Since we're *already* touching most of the repository...